ownership-chain/pallets/parachain-staking/src/lib.rs

```diff
@@ -1682,13 +1688,23 @@ pub mod pallet {
 			let target = ensure_signed(origin)?;
 
 			// reset rewards
-			let rewards = Rewards::<T>::take(&target);
-			ensure!(!rewards.is_zero(), Error::<T>::RewardsNotFound);
+			Rewards::<T>::try_mutate_exists(target.clone(), |maybe_rewards| {
+				let rewards = maybe_rewards.take().ok_or(Error::<T>::RewardsNotFound)?;
```

Comment:
can we here just do `take()`? then if rewards are not found it will fail in the next line as in the previous implementation

Reply:
but it's two different `take()` functions. first one is from `StorageMap` interface which removes the value from storage, and second one is from `Option` which takes the value out of the option https://doc.rust-lang.org/std/option/enum.Option.html#method.take